plugin/meshnet.go

```diff
@@ -414,6 +417,21 @@ func cmdDel(args *skel.CmdArgs) error {
 
 	meshnetClient := mpb.NewLocalClient(conn)
 
+	log.Infof("Del: Retrieving pod's (%s@%s) metadata from meshnet daemon", string(cniArgs.K8S_POD_NAME), string(cniArgs.K8S_POD_NAMESPACE))
+	localPod, err := meshnetClient.Get(ctx, &mpb.PodQuery{
+		Name:   string(cniArgs.K8S_POD_NAME), // getting deatils of the curretn pod.
+		KubeNs: string(cniArgs.K8S_POD_NAMESPACE),
+	})
+	if err != nil {
+		log.Infof("Del: Pod %s:%s is not in topology returning. err:%v", string(cniArgs.K8S_POD_NAMESPACE), string(cniArgs.K8S_POD_NAME), err)
+		return types.PrintResult(result, n.CNIVersion)
+	}
+	// if the current containerID is not the topo's current containerID, exit here b/c this is a duplicated DEL call.
+	if localPod.ContainerId != args.ContainerID {
+		log.Infof("Del: Pod %s:%s is a duplicate; skipping", string(cniArgs.K8S_POD_NAMESPACE), string(cniArgs.K8S_POD_NAME))
+		return nil
+	}
+
 	/* Tell daemon to close the grpc tunnel for this pod netns (if any) */
```

Comment:
This proposal is absolutely fine. 

An alternative could be, if we see the (localPod.NetNs == "" and localPod.SrcIp == "" ), this will also indicate that this is a duplicated DEL.  In this case we avoid adding container_id in the CRD.